mpack-instance-manager/src/main/python/instance_manager/instance_manager.py

```diff
@@ -0,0 +1,500 @@
+
+    os.symlink(mpack_path, target_link)
+    print "\nCreated new link " + target_link + " -> " + mpack_path
+
+  @staticmethod
+  def parse_into_component_instance_dict(path, component_names_filter=None):
+    result = {}
+    walk = os.walk(path)
+    for component_instance_name in next(walk)[1]:
+      if not component_names_filter or component_instance_name in component_names_filter:
+        result[component_instance_name] = ComponentInstance(name=component_instance_name,
+                                                            component_path=os.path.join(path, component_instance_name),
+                                                            path_exec=os.path.realpath(
+                                                              os.path.join(path, component_instance_name,
+                                                                           CURRENT_SOFTLINK_NAME)))
+    return result
+
+  @staticmethod
+  def parse_into_components_dict(path, components_filter, components_name_filter_map):
+    result = {}
+    walk = os.walk(path)
+    for component_type in next(walk)[1]:
+      if components_filter:
+        if components_filter == '*':
+          result[component_type] = ComponentInstance.parse_into_component_instance_dict(
+            os.path.join(path, component_type))
+        else:
+          component_instance_dict = ComponentInstance.parse_into_component_instance_dict(
+            os.path.join(path, component_type), components_filter)
+          if component_instance_dict:
+            result[component_type] = component_instance_dict
+      elif not components_name_filter_map:
+        result[component_type] = ComponentInstance.parse_into_component_instance_dict(
+          os.path.join(path, component_type))
+      elif component_type in components_name_filter_map.keys():
+        component_instance_dict = ComponentInstance.parse_into_component_instance_dict(
+          os.path.join(path, component_type), components_name_filter_map[component_type])
+        if component_instance_dict:
+          result[component_type] = component_instance_dict
+    return result
+
+  @staticmethod
+  def create_component_instance(mpack_name, mpack_version, mpack_instance, subgroup_name, module_name,
+                                component_type, component_instance_name, is_client_module):
+    if is_client_module:
+      component_path = os.path.join(ROOT_FOLDER_PATH, INSTANCES_FOLDER_NAME, mpack_name, mpack_instance, subgroup_name,
+                                    component_type)
+    else:
+      component_path = os.path.join(ROOT_FOLDER_PATH, INSTANCES_FOLDER_NAME, mpack_name, mpack_instance, subgroup_name,
+                                    module_name, component_type, component_instance_name)
+    mpack_path = os.path.join(ROOT_FOLDER_PATH, MPACKS_FOLDER_NAME, mpack_name, mpack_version, component_type)
+
+    if not os.path.lexists(mpack_path):
+      raise ValueError("Path doesn't exist: " + mpack_path)
```

Comment:
I assume it as 1st time creation of instance under the component type. What place is path created then ?